cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -335,4 +336,50 @@ void changeFlowletSteamConnection(Id.Program flow, String flowletId, String oldV
    * @return a list of all registered namespaces
    */
   List<NamespaceMeta> listNamespaces();
+
+  /**
+   * Creates an adapter if it does not exist.
```

Comment:
We don't throw an exception. It will update with the new value. Similar to how applications work.